SETUP/DEVELOPMENT.md

```diff
@@ -72,6 +72,50 @@ To install nodejs dependencies run the following in the repo base:
 npm install
 ```
 
+### JS development & deployment
+
+tl;dr: To develop JS: Delete `dist/manifest.php` if it exists. Either delete
+`dist/*` _or_ run `npm run build` whenever you change JS code.
+
+The site uses JS code only in the browser. This code uses ES6 modules which
+work directly in the browser (see [INSTALL.md](INSTALL.md) for the list of
+supported browsers). However, effectively using ES6 modules in production
+requires bundling the code such that all files have content-based hashes in
+the filenames to thwart browser caching when new code is released.
+
+The code uses [webpack](https://webpack.js.org/) to bundle the JS code and
+create a JSON manifest file that can be consumed by the PHP code. The JSON
+manifest file can then optionally be converted into a PHP file which is cached
```

Comment:
to avoid additional scenarios (json vs php), would it make sense to just use .js and cache the file in redis? Probably too much since redis isn't required and invalidating it for updates is probably a pain. Maybe we just don't support the .json flow? Just trying to reduce scenarios to test/support.

Reply:
As I went to take the JSON bit out I realized there's a development use-case for it we might want to consider. 

Without the JSON bit a JS developer is going to need to force-reload the page on every code update (or have the browser tools open to always have it always reload content). With the JSON bit, a JS developer can use `npx webpack --watch --progress` to auto-detect changes in the JS code and regenerate the JSON which is automatically picked up on the next PHP page load, so no need to force-refresh anything.

It seems like the latter is a more useful devex that removes the possibility that JS code is updated by the dev but not picked up by the browser. Thoughts?

Reply:
I could see that, but my development workflows for pgdp so far have not included the webpack server so I'd probably just continue that way. If you don't feel it is too much to carry all 3 scenarios, no need to remove since it is already present. Just an idea and me trying to understand each one.

Reply:
@70ray - as our most active JS developer, I'm very interested in your thoughts on this PR and the development workflow.